api/src/main/java/jakarta/data/repository/Find.java

```diff
@@ -55,23 +69,45 @@
  * instances with the given value of the {@code model} field.
  * </p>
  *
- * <p>A method annotated with {@code @Find} must return one of the following types:</p>
+ * <p>A method annotated with {@code @Find} must return one of the following types,
+ * where {@code E} is the entity type, entity attribute type, or record type inferred or explicitly specified by the {@code @Find} annotation:</p>
  * <ul>
- *     <li>an entity type {@code E}, when the method returns a single instance,</li>
+ *     <li>{@code E}, when the method returns a single instance</li>
  *     <li>{@code Optional<E>}, when the method returns at most a single instance,</li>
- *     <li>an entity array type {@code E[]},
+ *     <li>an array type {@code E[]},
  *     <li>{@code List<E>},</li>
- *     <li>{@code Stream<E>}, or</li>
- *     <li>{@code Page<E>} or {@code CursoredPage<E>}.</li>
+ *     <li>{@code Stream<E>},</li>
+ *     <li>{@code Page<E>}, or</li>
+ *     <li>{@code CursoredPage<E>} (only allowed when {@code E} is the entity type).</li>
  * </ul>
  *
- * <p>An automatic query method annotated {@code Find} returns an entity instance for every record which satisfies the
- * parameter-based conditions.</p>
+ * <p>For example, if a {@code Car} entity has attribute names including {@code make},
+ * {@code model}, {@code year}, and {@code vin}, a repository can use a Java record
+ * to request that only a subset of entity attributes be retrieved,</p>
+ * <pre>
+ * &#64;Repository
+ * public interface Cars extends BasicRepository&lt;Car, String&gt; {
+ *     record ModelInfo(String make,
+ *                      String model,
+ *                      int year) {}
+ *
+ *     &#64;Find
+ *     Optional&lt;ModelInfo&gt; getModelInfo(@By("vin") String vehicleIdNum);
```

Comment:
I enjoy the @gavinking approach on this find one:

https://github.com/jakartaee/data/pull/921#issuecomment-2569090819


```java
   @Find(Car.class) 
   @Project({_Car.MODEL, _Car.MAKE, _Car.YEAR})
   Optional<ModelInfo> getModelInfo(String vin);
```

Thus, I centralized the "builder" into a single place.